### DISCUSSION TITLE
Rule of thumb number of basins

### DISCUSSION THREAD

**Discussion by rlsandovalp** [neuralhydrology/neuralhydrology#128] [Q&A] [answered]
Hi, 

I see that there are some videos/publications/presentations where it is written "never train single basin LSTM", a bit late for me :/ 

However, I was wondering, is there any rule of thumb regarding the minimum number of basins that you should use for training? I suppose that the more the better, but if I want to make predictions on only one basin, are 3 basins enough before fine-tuning? 10? 100? 

Also, is there any recommendation about which basins should I use? Is it better to use basins of similar size to the basin for which I want to make predictions? basins of similar elevations?

Thanks!

**Comment by kratzert** [comment, marked as answer]
There is actually some experiments happening right now that might partially answer this question. Generally, the more the better and also the more _diverse_ the better (not only in size but also in terms of characteristics). It is good to have a good kind of mix of basins so that the model can really learn a general understanding of the underlying system. 

That being said, it is really hard to come up with the one golden number. I.e. in https://hess.copernicus.org/articles/26/3537/2022/hess-26-3537-2022-discussion.html we trained on  around 140 basins (iirc), all from the same region, so in some regards more similar than in other papers, but it was enough for this challenge. I would definitely recommend to use more than 3 or 10 and much rather aim for more than 100. However, if you are e.g. not able to get this amount of basins in your target region, you can also think about pretraining your model in a different region with a different dataset and then maybe finetune it to your region to the basins you have available.

Does this help?

**Comment by rlsandovalp** [comment]
Yes, it helps! Thanks!

Another question (maybe I can open a new discussion). In [this ](https://hess.copernicus.org/articles/26/1673/2022/) paper it is written that the problem with Bayesian approaches is that "any distribution that we could possibly assign is necessarily degenerate". I'm sorry I really tried to read the paper cited (Beven et al., 2008) but it was beyond my understanding. I don't understand what you mean by "degenerate". Does this mean that the uncertainty estimations rendered by a Bayesian neural network will always be wrong from a theoretical point of view? What makes it wrong?

Again, thanks a lot for the time you use to answer these questions!

**Reply by grey-nearing** [reply]
Hi Leonardo!  

Thanks so much for your insightful question :)

Daniel (first author on the paper you linked) asked me to field this question because I wrote the sentence that you are referring to. What we mean by "degenerate" is when a probability distribution that has support only on a projection of the true random variable space. This occurs, for example, when placing probability distributions over different models, since we can never test models that cover the entire space of possible models.  

This means that if we apply a chain rule expansion, e.g., through Bayesian inference, on a problem that includes uncertainty in the model itself, then any distribution we get at the end will not place finite (non-zero) probability on the "true" model. 

This does *not* mean that Bayesian or probabilistic inference will place zero probability on the actual value of any predicted variable. For example, we might have a continuous distribution that assigns finite probability to all possibly values of streamflow, even though we arrived at that predictive distribution using a degenerate distribution over model uncertainty. But it does mean that any predictive distribution we get will be wrong in a meaningful sense.

This is one of my favorite papers on this topic:  https://link.springer.com/article/10.1007/s10994-007-0716-7

What we are saying in that sentence is that we can skip the step of using any type of chain rule based probabilistic inference (and thus avoid any distribution over the model itself, which is the degenerate distribution), by asking the model to directly predict uncertainty. In the paper you are referencing, we are testing a discriminative model instead of a generative (e.g., Bayesian) model: https://www.sciencedirect.com/science/article/abs/pii/S0022169413007488.

We are *not* saying that a Bayesian neural network will always be wrong. We are saying that a chain rule inference that includes a distribution over models (like what Beven used) will always be wrong.

**Reply by rlsandovalp** [reply]
Thanks! very well explained :)
